### PR TITLE
Refactor/time strings

### LINKED ISSUE #14: Do not show warning message on tasks that were not created a long time ago
The bot shows a warning message on tasks that were opened recently, as seen [here](https://github.com/ubiquity/work.ubq.fi/issues/74#issuecomment-2242566257). It should only display this warning above a certain threshold, which comes from the configuration.

A possible cause would be that value missing in the current configuration. If that is the case, the default threshold should be set, and definitely above 0 days.

---

Tasks to be carried out:
- display a warning message when the task is above the configuration threshold
- do not display the warning if it is under that threshold
- change the configuration to accept a string representing a duration instead of a time stamp
- related tests

### PR DESCRIPTION
**Pull request by Keyrxng**
Resolves #14

> Tasks to be carried out:
> - related tests

I'm assuming this will be merged quickly and once it and #9 both are more specific test cases can be built if needed. I don't wish to replicate the same changes across PRs in regards to fixing the pr and issue fetching etc

Refactors:
- direct config access no need for `timers` and `misc` props
- `isTaskStale` uses `ms` same as config items and more readable

with config: 
```yml
      - plugin: http://localhost:4000
        runsOn: ["issue_comment.created"]
        with:
          reviewDelayTolerance: "3 Days"
          taskStaleTimeoutDuration: "15 Days"
          maxConcurrentTasks: 3
          startRequiresWallet: false
```

QA:

- [More than 15 days old](https://github.com/ubq-testing/bot-ai/issues/50)
- [Less than 15 days old](https://github.com/ubq-testing/bot-ai/issues/51)

### CODE REVIEW

File: src/types/plugin-input.ts [hunk 1]
```diff
@@ -11,24 +11,22 @@ export interface PluginInputs<T extends SupportedEventsU = SupportedEventsU, TU
...
-      startRequiresWallet: T.Boolean(),
+export const startStopSchema = T.Object(
+  {
+    reviewDelayTolerance: T.String({ default: "5 Days" }),
+    taskStaleTimeoutDuration: T.String({ default: "30 Days" }),
+    maxConcurrentTasks: T.Number({ default: 3 }),
+    startRequiresWallet: T.Boolean({ default: true }),
+  },
+  {
+    default: {
```
Comment (whilefoo) [inline src/types/plugin-input.ts:22]: you can leave this default as `{}`

File: src/handlers/shared/check-task-stale.ts [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+    return false;
   }
 
-  return false;
+  const currentDate = new Date();
+  const createdDate = new Date(createdAt);
+  const daysSinceCreation = calculateDaysDifference(currentDate, createdDate);
+  const staleDaysThreshold = Math.floor(staleTaskMilliseconds / ms("1d"));
+
+  return daysSinceCreation >= staleDaysThreshold && staleDaysThreshold > 0;
```
Comment (whilefoo) [inline src/handlers/shared/check-task-stale.ts:18]: is there a reason to convert to days? just get milliseconds of both dates and see if its higher than `staleTaskMilliseconds`

**Review by whilefoo — CHANGES_REQUESTED** [review]

File: src/utils/issue.ts [hunk 3]
```diff
@@ -183,13 +184,26 @@ export async function getAvailableOpenedPullRequests(context: Context, username:
       }
     }
 
-    if (reviews.length === 0 && (new Date().getTime() - new Date(openedPullRequest.created_at).getTime()) / (1000 * 60 * 60) >= reviewDelayTolerance) {
+    if (
+      reviews.length === 0 &&
+      (new Date().getTime() - new Date(openedPullRequest.created_at).getTime()) / (1000 * 60 * 60) >= getTimeValue(reviewDelayTolerance)
```
Comment (gentlementlegen) [inline src/utils/issue.ts:189]: Might wanna change `(1000 * 60 * 60)` tu use `ms`

File: README.md [hunk 4]
```diff
@@ -36,12 +36,10 @@ To configure your Ubiquibot to run this plugin, add the following to the `.ubiqu
...
   example: "/start" # or "/stop"
   with:
-    timers:
-      reviewDelayTolerance: 86000
-      taskStaleTimeoutDuration: 2580000
-    miscellaneous:
-      maxConcurrentTasks: 3
-      startRequiresWallet: true # default is true
+    reviewDelayTolerance: "3 Days"
+    taskStaleTimeoutDuration: "30 Days"
```
Comment (0x4007) [inline README.md:40]:
```suggestion
    reviewDelayTolerance: "1 Day"
    taskStaleTimeoutDuration: "7 Days"
```

Should be the defaults as a heads up.

File: src/types/plugin-input.ts [hunk 5]
```diff
@@ -11,24 +11,17 @@ export interface PluginInputs<T extends SupportedEventsU = SupportedEventsU, TU
...
-      maxConcurrentTasks: T.Number(),
-      startRequiresWallet: T.Boolean(),
-    },
-    { default: { maxConcurrentTasks: 3, startRequiresWallet: true } }
-  ),
-});
+export const startStopSchema = T.Object(
+  {
+    reviewDelayTolerance: T.String({ default: "5 Days" }),
+    taskStaleTimeoutDuration: T.String({ default: "30 Days" }),
```
Comment (0x4007) [inline src/types/plugin-input.ts:17]:
```suggestion
    reviewDelayTolerance: T.String({ default: "1 Day" }),
    taskStaleTimeoutDuration: T.String({ default: "1 Week" }),
```

**Review by 0x4007 — COMMENTED** [review]
I'm assuming:
- `reviewDelayTolerance`: how long we will give to the reviewers until we lift one limit from the contributor to start a new task. 
- `taskStaleTimeoutDuration`: I thought this is when the contributor gets unassigned but I'm not sure what this is in the context of the `/start` command.

**Comment by 0x4007** [comment]
Warning message to boot them? I thought that's a different plugin. 

Should be 3.5 days to warn and 7 to boot.

File: src/handlers/shared/check-task-stale.ts [hunk 6]
```diff
@@ -1,9 +1,11 @@
...
+  if (staleTaskMilliseconds === 0) {
+    return false;
   }
 
-  return false;
+  const currentDate = new Date();
+  const createdDate = new Date(createdAt);
+  const millisecondsSinceCreation = currentDate.getTime() - createdDate.getTime();
+
+  return millisecondsSinceCreation >= staleTaskMilliseconds && staleTaskMilliseconds > 0;
```
Comment (gentlementlegen) [inline src/handlers/shared/check-task-stale.ts:10]: Maybe this check is redundant since you return `false` when `staleTaskMilliseconds` is equal to zero

**Review by gentlementlegen — APPROVED** [review]

File: src/utils/issue.ts [hunk 7]
```diff
@@ -183,13 +184,23 @@ export async function getAvailableOpenedPullRequests(context: Context, username:
       }
     }
 
-    if (reviews.length === 0 && (new Date().getTime() - new Date(openedPullRequest.created_at).getTime()) / (1000 * 60 * 60) >= reviewDelayTolerance) {
+    if (reviews.length === 0 && (new Date().getTime() - new Date(openedPullRequest.created_at).getTime()) / ms("1h") >= getTimeValue(reviewDelayTolerance)) {
```
Comment (whilefoo) [inline src/utils/issue.ts:187]: why is it converted to hours if `getTimeValue(reviewDelayTolerance)` returns milliseconds or am I missing something

**Review by whilefoo — CHANGES_REQUESTED** [review]

**Comment by Keyrxng** [comment]
Bumping this for review as it also now resolves the issue exposed with the `LogReturn` logger export used in the bubble-up error comment

**Review by whilefoo — APPROVED** [review]

File: src/handlers/shared/generate-assignment-comment.ts [hunk 8]
```diff
@@ -15,11 +15,11 @@ export async function generateAssignmentComment(context: Context, issueCreatedAt
   let endTime: null | Date = null;
   let deadline: null | string = null;
   endTime = new Date(startTime + duration * 1000);
-  deadline = endTime.toLocaleString("en-US", options);
+  deadline = endTime.toLocaleString("en-US", options)
```
Comment (gentlementlegen) [inline src/handlers/shared/generate-assignment-comment.ts:18]: @Keyrxng I believe this is not intended?

Comment (Keyrxng) [reply src/handlers/shared/generate-assignment-comment.ts:18]: Oop yeah not intended, forgot to run format after deciding to adjust the deadline elsewhere

Comment (gentlementlegen) [reply src/handlers/shared/generate-assignment-comment.ts:18]: @Keyrxng Can you please fix this? I am fine with the PR otherwise 😄

**Review by gentlementlegen — APPROVED** [review]

**Pull request merged**